src/utils/invariant.ts

```diff
@@ -0,0 +1,15 @@
+const genericMessage = "Invariant Violation";
+
+export class InvariantError extends Error {
+  framesToPop = 1;
+  name = genericMessage;
+  constructor(message: string = genericMessage) {
+    super(message);
+  }
+}
+
+export default function invariant(condition: any, message: string) {
+  if (!condition) {
+    throw new InvariantError(message);
```

Comment:
Although we're not using `InvariantError` directly outside of this module yet, this line of code gets transformed to
```ts
throw process.env.NODE_ENV === "production" ? new InvariantError(message) : new InvariantError();
```
which is fine/harmless… and even saves a couple bytes in production!